nhrpd/nhrp_vty.c

```diff
@@ -594,29 +595,60 @@ struct info_ctx {
 	struct vty *vty;
 	afi_t afi;
 	int count;
+	struct json_object *json;
 };
 
 static void show_ip_nhrp_cache(struct nhrp_cache *c, void *pctx)
 {
 	struct info_ctx *ctx = pctx;
 	struct vty *vty = ctx->vty;
 	char buf[2][SU_ADDRSTRLEN];
+	struct json_object *json = NULL;
 
 	if (ctx->afi != family2afi(sockunion_family(&c->remote_addr)))
 		return;
 
-	if (!ctx->count) {
+	if (ctx->json)
+		json = json_object_new_object();
+	if (!ctx->count && !ctx->json) {
 		vty_out(vty, "%-8s %-8s %-24s %-24s %-6s %s\n", "Iface", "Type",
 			"Protocol", "NBMA", "Flags", "Identity");
 	}
 	ctx->count++;
 
+	sockunion2str(&c->remote_addr, buf[0], sizeof buf[0]);
+	if (c->cur.peer)
+		sockunion2str(&c->cur.peer->vc->remote.nbma,
+			      buf[1], sizeof buf[1]);
+	else
+		snprintf(buf[1], sizeof buf[1], "-");
+	if (json) {
+		json_object_string_add(json, "Interface", c->ifp->name);
```

Comment:
Please follow camelCase

`interface`
`type`
`protocol`

first word is lowercase second word and beyond is uppercase